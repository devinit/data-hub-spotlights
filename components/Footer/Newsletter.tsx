import React, { FunctionComponent } from 'react';
import { Newsletter as NewsletterType } from '../DefaultLayout';

type NewsletterProps = NewsletterType;

const Newsletter: FunctionComponent<NewsletterProps> = ({ caption, link_label, link_url }) => {
  return (
    <>
      <h4 className="footer__title">{ caption }</h4>
      <a href={ link_url } className="button">{ link_label }</a>
    </>
  );
};

export { Newsletter };
