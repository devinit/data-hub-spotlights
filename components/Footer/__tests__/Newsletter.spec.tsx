/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { Newsletter } from '../Newsletter';

describe('Newsletter', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(
      <Newsletter caption="Sign up to our newsletter" link_label="Sign Up" link_url="https://localhost:3000/newsletter"/>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
