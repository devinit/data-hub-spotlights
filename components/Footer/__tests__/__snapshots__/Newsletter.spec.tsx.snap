// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsletter renders correctly 1`] = `
Array [
  <h4
    className="footer__title"
  >
    Sign up to our newsletter
  </h4>,
  <a
    className="button"
    href="https://localhost:3000/newsletter"
  >
    Sign Up
  </a>,
]
`;
